api-client-android-oauth/src/main/java/com/xing/api/oauth/XingOAuth.java

```diff
@@ -82,16 +95,44 @@ public void loginWithXing(Fragment fragment) {
         startLoginForCaller(fragment);
     }
 
-    /**  */
-    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+    /**
+     * Starts the "Login with XING" process, by showing an auth screen.
+     * <p>
+     * This calls the {@linkplain Fragment#startActivityForResult} method.
+     */
+    public void loginWithXing(android.support.v4.app.Fragment fragment) {
+        startLoginForCaller(fragment);
+    }
+
+    /**
+     * Parses the activity result and deferments if the response was from a {@linkplain XingOAuthActivity}.
+     * Returns a {@linkplain OAuthResponse} with the authentication result, otherwise {@code null}.
+     */
+    @Nullable
+    public OAuthResponse onActivityResult(int requestCode, int resultCode, Intent data) {
         if (requestCode == Shared.REQUEST_CODE) {
             if (resultCode == Activity.RESULT_OK) {
                 Bundle extras = data.getExtras();
-                callback.onSuccess(extras.getString(Shared.TOKEN, ""), extras.getString(Shared.TOKEN_SECRET, ""));
-            } else if (resultCode == Activity.RESULT_CANCELED) {
+                String token = extras.getString(Shared.TOKEN, "");
+                String tokenSecret = extras.getString(Shared.TOKEN_SECRET, "");
+                // FIXME: (2.1.0) To be removed. Calling callback to maintain backwards compatibility.
+                if (callback != null) {
+                    callback.onSuccess(extras.getString(Shared.TOKEN, ""), extras.getString(Shared.TOKEN_SECRET, ""));
+                }
+                // Return success response
+                return OAuthResponse.success(token, tokenSecret);
+            }
+
+            // FIXME: (2.1.0) To be removed. Calling callback to maintain backwards compatibility.
+            if (callback != null) {
                 callback.onError();
             }
+            // TODO: (2.1.0) Parse error reason to provide feedback for the user.
+            // Return error response
+            return OAuthResponse.error("Aborted");
         }
+
+        return null;
```

Comment:
This will require us to introduce a third state, which may not be clear for the end user. In the end the user will end-up doing two checks. So it's:
```
  if (oauthResponse != null) {
  }
```
vs
```
  if (oauthResponse != OAuthResponse.EMPTY) {
  }
```
the later would mean, that `OAuthResponse.EMPTY` would have to return a `false` on `isSuccessful()`, but what actually happened is that the `onActivityResult()` was called after coming back from a different activity, not ours.

In this case `null` makes more sense, means that the invocation of the method, is not related to us.